src/monodroid/jni/monodroid-glue.cc

```diff
@@ -1924,7 +1945,12 @@ Java_mono_android_Runtime_init (JNIEnv *env, jclass klass, jstring lang, jobject
 	char *aotMode;
 	int i;
 
-	init_logging_categories ();
```

Comment:
I don't think we should remove this call.

The `init_logging_categories()` invocation in `JNI_OnLoad()` allows *global* settings to be processed, but not per-package settings.  This *is* noted in the comment in `JNI_OnLoad()`, but this is basically removal of existing functionality.

If we call `init_logging_categories()` *twice* -- once in `JNI_OnLoad()` and once here -- then existing functionality is retained, while also allowing "global" state to be used before `Java_mono_android_Runtime_init()` is invoked.

It should also be safe to call `init_logging_categories()` twice; at least, I don't see anything obviously wrong with doing so.

Reply:
Calling it twice means parsing stuff twice, which means everything gets a bit slower...

Reply:
I just removed it from `JNI_OnLoad`, together with all the timing code